Content.Server/GameTicking/Rules/ZombieRuleSystem.cs

```diff
@@ -130,6 +133,26 @@ private void CheckRoundEnd(ZombieRuleComponent zombieRuleComponent)
             _roundEnd.EndRound();
     }
 
+    /// <summary>
+    ///     Doesn't let to recall the shuttle. No roundstalling
+    /// </summary>
+    private void OnShuttleCallAttempt(ref CommunicationConsoleCallShuttleAttemptEvent ev)
+    {
+        var query = QueryActiveRules();
+        while (query.MoveNext(out _, out _, out var zombie, out _))
+        {
+            if (zombie is { ZombieShuttleCalled: false })
+            {
```

Comment:
Is this part necessary? Currently it's a bug as it's checking if it's false and then true (which is impossible).

Reply:
well it was basically copied from
https://github.com/space-wizards/space-station-14/blob/5958801182b35bb1e78095f18b180d3069ad7394/Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs#L305-L323

as i understand i get ZombieGameRule entity only this way

Reply:
Nukeops does is because it is trying to pattern match if the field is not null, but we don't have to do this because we're using a boolean and not a nullable value.

As we're just working with a boolean, we can access it directly:

```cs
    private void OnShuttleCallAttempt(ref CommunicationConsoleCallShuttleAttemptEvent ev)
    {
        var query = QueryActiveRules();
        while (query.MoveNext(out _, out _, out var zombie, out _))
        {
            if (!zombie.ZombieShuttleCalled)
                continue;

            ev.Cancelled = true;
            ev.Reason = Loc.GetString(zombie.ShuttleCallUnavailableReason);
            return;
        }
    }

```

The current pattern matching is incorrect as it first checks that it is false then it checks that it is true, which is impossible outside of something like a race condition (true != false).

Some examples in other parts of the codebase:
https://github.com/space-wizards/space-station-14/blob/2a6800f35411cf35213f2ef1dfecf2e5c4aa9ed5/Content.Server/Antag/AntagSelectionSystem.cs#L85-L106
https://github.com/space-wizards/space-station-14/blob/2a6800f35411cf35213f2ef1dfecf2e5c4aa9ed5/Content.Server/GameTicking/Rules/RuleGridsSystem.cs#L28-L39

Reply:
will know in the future, thanks!